packages/Webkul/Product/src/Repositories/ProductCustomerGroupPriceRepository.php

```diff
@@ -33,10 +35,27 @@ public function saveCustomerGroupPrices(array $data, $product)
                     $row['customer_group_id'],
                 ]));
 
+                if (in_array($row['unique_id'], $seenUniqueIds)) {
+                    throw new \Exception(trans('admin::app.catalog.products.edit.price.group.duplicate-error'));
+                }
+                $seenUniqueIds[] = $row['unique_id'];
+
                 if (Str::contains($customerGroupPriceId, 'price_')) {
-                    $this->create(array_merge([
-                        'product_id' => $product->id,
-                    ], $row));
+                    $existingPrice = $this->findOneWhere(['unique_id' => $row['unique_id']]);
+                    if ($existingPrice) {
+                        throw new \Exception(trans('admin::app.catalog.products.edit.price.group.duplicate-error'));
```

Comment:
This should send a code and code should be added as constant with proper doc comment.

Because this will break when tranlation changed because you have added check with translation itsellf.